.travis.yml

```diff
@@ -9,48 +9,49 @@ services:
 
 jobs:
   include:
-    - stage: e2e
+    - stage: "build, unit test"
+      script: make
+    - stage: "e2e"
       before_script:
-      - KIND_VERSION=0.4.0
-      - KUSTOMIZE_VERSION=3.0.2
-      # Download and install kind
-      - curl -L https://github.com/kubernetes-sigs/kind/releases/download/v${KIND_VERSION}/kind-linux-amd64 --output kind && chmod +x kind && sudo mv kind /usr/local/bin/
-      # Create kind cluster
-      - |
-        if [ $(kind get clusters) ]; then
-          export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
-        else
-          kind create cluster
-        fi
-      # Get kubeconfig
-      - export KUBECONFIG="$(kind get kubeconfig-path --name="kind")"
-      # Build image and tag as gatekeeper-e2e:latest
-      - make docker-build REPOSITORY=gatekeeper-e2e
-      # Load Gatekeeper image into kind cluster
-      - kind load docker-image --name kind gatekeeper-e2e:latest
-      # Download and install kubectl
-      - curl -LO https://storage.googleapis.com/kubernetes-release/release/$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)/bin/linux/amd64/kubectl && chmod +x ./kubectl && sudo mv kubectl /usr/local/bin/
-      # Download and install kustomize
-      - curl -L https://github.com/kubernetes-sigs/kustomize/releases/download/v${KUSTOMIZE_VERSION}/kustomize_${KUSTOMIZE_VERSION}_linux_amd64 --output kustomize && chmod +x kustomize && sudo mv kustomize /usr/local/bin/
+      - make e2e-bootstrap
+      - export KUBECONFIG="$(kind get kubeconfig-path --name="kind")" 
+      - make e2e-build-load-image IMG=gatekeeper-e2e:latest
+      - make patch-image IMG=gatekeeper-e2e:latest USE_LOCAL_IMG=true
       script:
-      - make deploy
+      - make deploy 
       - make test-e2e
       - echo -e '\n\n======= manager logs =======\n\n' && kubectl logs -n gatekeeper-system gatekeeper-controller-manager-0 manager
-    - stage: "build, test, deploy"
-      script: make
+    - stage: "dev deploy"
+      if: repo = open-policy-agent/gatekeeper AND type != pull_request
+      script: 
+      - make manager
       deploy:
```

Comment:
Because we're relying on `deploy` to gate whether this runs () `branch: master` and `tags: false`, we can move this to the `build, unit test` stage to avoid needing build twice.